core/src/navigation/services/context-switcher.js

```diff
@@ -62,7 +63,11 @@ export const ContextSwitcherHelpers = {
     }
 
     // we are inside the context switcher base path
-    const selectedId = currentPath.replace(parentNodePath, '').split('/')[0];
+    const selectedId = currentPath
+      .replace(parentNodePath, '')
+      .split('/')[0]
+      .split(nodeParamPrefix)[0] //ignore everything after nodeParamPrefix
```

Comment:
I would remove .split(nodeParamPrefix)[0] and just let the .split('?')[0] there. There is no rule that forbids nodeParamPrefix to be a character not allowed in a path segment.

Reply:
Would you like to see this?
![image](https://user-images.githubusercontent.com/22825433/51036531-aac3e080-15ad-11e9-8c31-acd76fe89e5b.png)

Reply:
yes!

Reply:
Allright then...